src/decode.rs

```diff
@@ -5065,6 +5059,7 @@ pub unsafe fn rav1d_submit_frame(c: &mut Rav1dContext) -> Rav1dResult {
                     "Compiled without support for {}-bit decoding",
                     8 + 2 * seq_hdr.hbd
                 );
+                drop(task_thread_lock);
                 on_error(f, c, out_delayed);
                 return Err(ENOPROTOOPT);
```

Comment:
General Q: Why do we explicitly drop the thread lock guard right before early returns? Wouldn't the guard's destructor get run on the return anyway? Is there something I'm missing or misinterpreting here?

Reply:
https://github.com/memorysafety/rav1d/pull/673#discussion_r1449506993